python/ray/train/v2/_internal/exceptions.py

```diff
@@ -86,6 +86,14 @@ class WorkerGroupStartupFailedError(RayTrainError):
     """
 
 
+class InsufficientClusterResourcesError(RayTrainError):
+    """Exception raised when the cluster has insufficient resources.
+
+    Example scenario: A worker that requires 1 GPU is scheduled onto a node that
```

Comment:
nit: `scheduled onto a cluster that only has 1 cpu` would be more accurate. The current wording implies that if a cluster has 1 node with 1 gpu and another node with 1 cpu it would still attempt to place the worker on the 1 cpu node.